include/sundials/sundials_config.in

```diff
@@ -31,6 +31,18 @@
 #  define SUNDIALS_DEPRECATED_NO_EXPORT_MSG(msg) SUNDIALS_NO_EXPORT SUNDIALS_DEPRECATED_MSG(msg)
 #endif
 
+/* ------------------------------------------------------------------
+ * Define C version check macros
+ * -----------------------------------------------------------------*/
+
+#if defined(__STDC__)
```

Comment:
Can we rely on `__STDC__` to always be defined (e.g., with [MSVC](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170) is seems like this could be undefined)? Could using `__STDC__` cause a problem with C++ codes (e.g., with how `SUNDIALS_C_INLINE` is defined below)?

Reply:
According to the docs, MSVC does not define `__STDC__` when using MSVC extensions (unless MSVC 2022 17.2, then it does also when c11 or c17 is used).  So I suppose we should add

```c
#elif _MSC_VER
#define SUNDIALS_STDC_VERSION 198901L
#else
#define SUNDIALS_STDC_VERSION 0 // unknown
#endif
```

Interesting that it works in the CI, I wonder if its disabling the MSVC extensions (which results in `__STDC__` being defined according to the docs).

Reply:
Most C++ compilers define `__STDC__` still, in fact there was a recent [LLVM RFC](https://discourse.llvm.org/t/rejected-rfc-stop-defining-the-stdc-and-related-macros-in-c-mode/62468/14) that wanted to change LLVM to not define it in C++ and it was rejected. GNU defines it also. However, `__STDC_VERSION__` is not defined (at least by GNU) when compiling for C++. But then again, is it even valid to pass a C++ compiler as the C compiler? We should probably not define `SUNDIALS_STDC_VERSION` if C++.

Reply:
I've had CMake throw an error before when trying to use a C++ compiler as the C compiler but I'm not sure if that's always the case. Some users in the past have built sundials using a C++ compiler (I think through their own application build system). Not sure that's a use case we should support though.

I was more thinking about a case where the C compiler defines `__STDC__` but the C++ compiler does not so `SUNDIALS_C_INLINE` could be defined differently between the two but I guess that's really a question about what the preprocessor does.